pkg/cli/admin/inspect/proxy.go

```diff
@@ -0,0 +1,61 @@
+package inspect
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+
+	"k8s.io/cli-runtime/pkg/resource"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+var proxyRegex = regexp.MustCompile(`(https?://)([^:]+):([^@]+)@`)
+
+type proxyList struct {
+	*configv1.ProxyList
+}
+
+func (c *proxyList) addItem(obj interface{}) error {
+	structuredItem, ok := obj.(*configv1.Proxy)
+	if !ok {
+		return fmt.Errorf("unhandledStructuredItemType: %T", obj)
+	}
+	c.Items = append(c.Items, *structuredItem)
+	return nil
+}
+
+func inspectProxyInfo(info *resource.Info, o *InspectOptions) error {
+	structuredObj, err := toStructuredObject[configv1.Proxy, configv1.ProxyList](info.Object)
+	if err != nil {
+		return err
+	}
+
+	switch castObj := structuredObj.(type) {
+	case *configv1.Proxy:
+		elideProxy(castObj)
+
+	case *configv1.ProxyList:
+		for i := range castObj.Items {
+			elideProxy(&castObj.Items[i])
+		}
+	}
+
+	// save the current object to disk
+	dirPath := dirPathForInfo(o.DestDir, info)
+	filename := filenameForInfo(info)
+	// ensure destination path exists
+	if err := os.MkdirAll(dirPath, os.ModePerm); err != nil {
+		return err
+	}
+	return o.fileWriter.WriteFromResource(path.Join(dirPath, filename), structuredObj)
+}
+
+// elideProxy obfuscates the sensitive information from proxy object.
+func elideProxy(proxy *configv1.Proxy) {
```

Comment:
Can you also unit test `elideProxy` or `inspectProxyInfo`?

Reply:
Added.